docs/product/specs/permissions-revamp.md

```diff
@@ -0,0 +1,146 @@
+  * Building our permission layer on top of PostgreSQL's permission system is future-proof and robust.
+- B. We should try to maintain feature parity with the flows we currently have.
+- C. Provide a rudimentary UX for ownerships and get users aquainted with the concept.
+
+## Features we intend to remove
+* We will remove the ability to share credentials across databases.
+  - We currently have this while creating/configuring a database.
+  - We will revisit this feature later down the road when we have requirements from users.
+
+## Terminologies
+* User refers to Mathesar user.
+* Admin refers to Mathesar admins.
+* Role refers to PostgreSQL role/user.
+
+## What we want the user to be able to do
+### Adding/configuring a database
+* A database should be unique. The same database cannot be added multiple times.
+  - This is a change from the existing setup where a database can be added multiple times with different connection credentials.
+* We'll have separate onboarding instructions/flows in the UI for pre-existing and new databases.
+
+#### Pre-existing database
+* Mathesar admins should be able to configure a pre-existing database, using the following details:
+  - DB server Host
+  - DB server Port
+  - Database name
+  - PostgreSQL role -> username and password
+* Mathesar admins should be able to optionally generate pre-defined roles for Editor, and Viewer, to partially satisfy Goal (B).
+  - This requires the PostgreSQL role used in the previous step to contain CREATEROLE attribute on the database server.
+  - This step is optional and Mathesar should function normally without these roles.
+  - We should make it clear that these roles & their privileges would only apply for new objects created via the Mathesar UI.
+  - For existing objects, we'll provide an option to iteratively provide privileges to these roles.
+    - This will however depend on ownerships of each object and can fail.
+  - We will no longer have the manager role as it would be the same as having admin access.
+    - Future consideration: We could provide a concept of database admins as a Mathesar level access control option per database. This will help maintain feature parity with the 'DB level admins' feature we currently have.
+#### New database
+* Mathesar admins should be able to create a new database on the internal database server by only entering the database name.
+  - This will use the same role utilized by the Django database.
+    - The user should be able to create new roles on the internal db and use them if they wanted to.
+  - The role used should contain both CREATEDATABASE and CREATEROLE privileges.
+* Mathesar admins should be able to create a new database on external servers, by specifying the same details as the 'pre-existing' database configuration.
+  - The role used should contain both CREATEDATABASE and CREATEROLE privileges.
```

Comment:
What does `CREATEDATABASE` even mean as a privilege if roles are subordinate to databases (in the UI, and in users' minds)? I.e., how is an admin user supposed to think about this if we're avoiding the concept of a server that has both roles and databases?

Reply:
This is no longer relevant after https://github.com/mathesar-foundation/mathesar-wiki/pull/110/commits/189b77285da6da17d2179f28c83d6233d2d557af